src/llscan.cc

```diff
@@ -289,6 +295,173 @@ bool NodeInfoCmd::DoExecute(SBDebugger d, char** cmd,
+
+  std::string full_cmd;
+  for (; start != nullptr && *start != nullptr; start++) full_cmd += *start;
+
+  SBExpressionOptions options;
+  SBValue value = target.EvaluateExpression(full_cmd.c_str(), options);
+
+  // Load V8 constants from postmortem data
+  llv8.Load(target);
+
+  if (value.GetError().Fail()) {
+    SBStream desc;
+    if (value.GetError().GetDescription(desc)) {
+      result.SetError(desc.GetData());
+    }
+    result.SetStatus(eReturnStatusFailed);
+    return false;
+  }
+
+  // Check the address we've been given at least looks like a valid object.
+  v8::Value search_value(&llv8, value.GetValueAsSigned());
+  v8::Smi smi(search_value);
+  if (smi.Check()) {
+    result.SetError("Search value is an SMI.");
+    return false;
+  }
+
+  // Walk all the object instances and handle them according to their type.
+  TypeRecordMap mapstoinstances = llscan.GetMapsToInstances();
+  for (auto const entry : mapstoinstances) {
+    TypeRecord* typerecord = entry.second;
+    for (uint64_t addr : typerecord->GetInstances()) {
+      v8::Error err;
+      v8::Value obj_value(&llv8, addr);
+      v8::HeapObject heap_object(obj_value);
+      int64_t type = heap_object.GetType(err);
+      v8::LLV8* v8 = heap_object.v8();
+
+      // We only need to handle the types that are in
+      // FindJSObjectsVisitor::IsAHistogramType
+      // as those are the only objects that end up in GetMapsToInstances
+      if (type == v8->types()->kJSObjectType ||
+          type == v8->types()->kJSArrayType) {
+        // Objects can have elements and arrays can have named properties.
+        // Basically we need to access objects and arrays as both objects and
+        // arrays.
+        v8::JSObject js_obj(heap_object);
+        PrintRefs(result, js_obj, search_value, err);
+
+      } else if (type < v8->types()->kFirstNonstringType) {
+        v8::String str(heap_object);
+        PrintRefs(result, str, search_value, err);
+
+      } else if (type < v8->types()->kJSTypedArrayType) {
+        // These should only point to off heap memory,
+        // this case should be a no-op.
+      } else {
+        // result.Printf("Unhandled type: %" PRId64 " for addr %" PRIx64
+        //    "\n", type, addr);
+      }
+    }
+  }
+
+  result.SetStatus(eReturnStatusSuccessFinishResult);
+  return true;
+}
+
+void FindReferencesCmd::PrintRefs(SBCommandReturnObject& result,
+                                  v8::JSObject& js_obj, v8::Value& search_value,
+                                  v8::Error& err) {
+  v8::Value::InspectOptions inspect_options;
+  inspect_options.detailed = false;
+  int64_t length = js_obj.GetArrayLength(err);
+  for (int64_t i = 0; i < length; ++i) {
+    v8::Value v = js_obj.GetArrayElement(i, err);
+    if (err.Success() && v.raw() == search_value.raw()) {
+      std::string type_name = js_obj.GetTypeName(&inspect_options, err);
+      result.Printf("0x%" PRIx64 ": %s[%" PRId64 "]=0x%" PRIx64 "\n",
```

Comment:
This could be more generic: instead of printing reference in-line - the callback may be passed to the `PrintRefs` functions. I'm not sure what arguments it should have, though... Perhaps `std::string` with object description and a `v8::Value`?

Doing it in a such way may be helpful in the future, if we would want to create tools for complicated analysis.